### PR TITLE
Feature/bunny experiment

### PR DESCRIPTION
**Pull request by denis-yuen**
Allow cwlavro to run with bunny as an alternative to cwltool

### CODE REVIEW

File: cwlavro-tools/src/main/java/io/cwl/avro/CWL.java [hunk 1]
```diff
@@ -14,18 +15,31 @@
...
+import org.apache.commons.compress.archivers.ArchiveException;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang3.tuple.ImmutablePair;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.IOException;
+import java.io.*;
```
Comment (garyluu) [inline cwlavro-tools/src/main/java/io/cwl/avro/CWL.java]: Should probably avoid importing *

File: cwlavro-tools/src/main/java/io/cwl/avro/CWL.java [hunk 2]
```diff
@@ -14,18 +15,31 @@
 import com.google.gson.JsonParseException;
 import com.google.gson.JsonPrimitive;
 import com.google.gson.reflect.TypeToken;
+import org.apache.commons.compress.archivers.ArchiveException;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
```
Comment (garyluu) [inline cwlavro-tools/src/main/java/io/cwl/avro/CWL.java]: See if lang3.StringUtils could be used instead

File: cwlavro-tools/src/main/java/io/cwl/avro/CWL.java [hunk 3]
```diff
@@ -38,11 +52,60 @@
  */
 public class CWL {
 
+    public static final String RABIX_VERSION = "1.0.0-rc5";
+    public static final String RABIX_GITHUB_LOCATION = "https://github.com/rabix/bunny/releases/download/v"+RABIX_VERSION+"/rabix-"+RABIX_VERSION+".tar.gz";
+    public static final String RABIX_EXEC_LOCATION = ".dockstore/libraries/rabix-"+RABIX_VERSION+"/rabix-backend-local-"+RABIX_VERSION+"/rabix";
+
     private final Gson gson;
     private static final Logger log = LoggerFactory.getLogger(CWL.class);
```
Comment (garyluu) [inline cwlavro-tools/src/main/java/io/cwl/avro/CWL.java]: General styling might be nice (spacing, static final all caps, etc)

File: cwlavro-tools/src/main/java/io/cwl/avro/CompressionUtilities.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                        if (!outputFile.mkdirs()) {
+                            throw new IllegalStateException(String.format("Couldn't create directory %s.", outputFile.getAbsolutePath()));
+                        }
+                    }
+                } else {
+                    LOG.info(String.format("Creating output file %s.", outputFile.getAbsolutePath()));
+                    FileUtils.forceMkdirParent(outputFile);
+                    final OutputStream outputFileStream = new FileOutputStream(outputFile);
+                    IOUtils.copy(debInputStream, outputFileStream);
+                    outputFileStream.close();
```
Comment (garyluu) [inline cwlavro-tools/src/main/java/io/cwl/avro/CompressionUtilities.java]: Try with resources for the OutputStream might be safer.

File: cwlavro-tools/src/main/java/io/cwl/avro/CWL.java (hunk 3 above)
Comment (denis-yuen) [inline cwlavro-tools/src/main/java/io/cwl/avro/CWL.java]: I don't think we have a checkstyle file setup for this dependency. 
That said, I think the formatting matches Dockstore already https://github.com/ga4gh/dockstore/blob/develop/dockstore-client/src/main/java/io/dockstore/client/cli/Client.java#L88

File: cwlavro-tools/src/main/java/io/cwl/avro/CWL.java (hunk 2 above)
Comment (denis-yuen) [inline cwlavro-tools/src/main/java/io/cwl/avro/CWL.java]: Good idea

File: cwlavro-tools/src/main/java/io/cwl/avro/CompressionUtilities.java (hunk 4 above)
Comment (denis-yuen) [inline cwlavro-tools/src/main/java/io/cwl/avro/CompressionUtilities.java]: :+1:

**Pull request merged**